packages/gatsby-source-drupal/src/gatsby-node.js

```diff
@@ -54,10 +65,49 @@ exports.sourceNodes = async (
     disallowedLinkTypes,
     skipFileDownloads,
     fastBuilds,
+    translation,
     entityReferenceRevisions = [],
   } = pluginOptions
   const { createNode, setPluginStatus, touchNode } = actions
 
+  // Default apiBase to `jsonapi`
+  apiBase = apiBase || `jsonapi`
+
+  // Default disallowedLinkTypes to self, describedby.
+  disallowedLinkTypes = disallowedLinkTypes || [`self`, `describedby`]
+
+  // Default concurrentFileRequests to `20`
+  concurrentFileRequests = concurrentFileRequests || 20
+
+  // Default skipFileDownloads to false.
+  skipFileDownloads = skipFileDownloads || false
+
+  // Determine what entities can be translated and what languages are enabled.
+  let languageConfig = store.getState().status.plugins?.[`gatsby-source-drupal`]
+    ?.languageConfig
+
+  if (!languageConfig) {
+    languageConfig = await fetchLanguageConfig({
+      translation,
+      baseUrl,
+      apiBase,
+      basicAuth,
+      headers,
+      params,
+      reporter,
+    })
+
+    if (translation && !languageConfig.enabledLanguages.length) {
+      reporter.warn(
+        `The translation option is enabled but no enabled languages were retrieved from Drupal. Make sure your basicAuth credentials in the Gatsby config file are populated with a user that has the "administer languages" permission.`
```

Comment:
It's important to note that because of this automatic language resolution devs are required to either:
- add `administer languages` permission to the anonymous user role which is not recommended
- add `basicAuth` parameter on the Gatsby side for a Drupal user who has a role with `administer languages`

Both of these are suboptimal but make the plugin a bit more plug and play. We could instead allow the dev a way to provide a list of languages and entity types for translation. What do you think? @KyleAMathews

Dropping the automatic language settings resolution would mean less code too which is always nice :)

Reply:
implemented in https://github.com/gatsbyjs/gatsby/pull/26720/commits/a518bcafbd6c61c90ac093bc6a31fe0333117efb